### PR TITLE
all auth

### PR DESCRIPTION
**Pull request by wgyt**

### CODE REVIEW

**Comment by wgyt** [comment]
@RedGuy12 it should work on replit, but because it uses repl db it doesm't work on gitpod

**Comment by wgyt** [comment]
wait, it doesn't work because it gives a 500 empty page error

File: index.js [hunk 1]
```diff
@@ -5,46 +5,39 @@ module.exports = async (req, res, config) => {
...
-					{ code: code, url: url },
-					{},
-					function (err, str) {
-						if (err) {
-							throw err;
-						}
-						text = str;
-					},
-				);
+				ejs.renderFile(`./pages/authwith${cloud}.ejs`, { code: code, url: url }, {}, function (err, str) {
```
Comment (RedGuy12) [inline index.js:21]: Codacy has a fix for the issue: [Expected property shorthand.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
				ejs.renderFile(`./pages/authwith${cloud}.ejs`, { code, url: url }, {}, function (err, str) {
```

Comment (RedGuy12) [reply index.js:21]: i never said that that was a bot

Comment (wgyt) [reply index.js:21]: that isn't the issue, this issue is the pages that get the key return empty for some dumb reason.

File: 1host.config.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+	port: 3000,
+	modules: [
+		{
+			module: require("./index.js"),
+			errorHandler: false,
+			config: { usernamee: "wgyt", projectid: "513842800", commentid: "200363559" },
```
Comment (RedGuy12) [inline 1host.config.js:7]: Codacy has a fix for the issue: [There should be no space after '{'.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
			config: {usernamee: "wgyt", projectid: "513842800", commentid: "200363559" },
```

File: 1host.config.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+	port: 3000,
+	modules: [
+		{
+			module: require("./index.js"),
+			errorHandler: false,
```
Comment (RedGuy12) [inline 1host.config.js:6]: Codacy has a fix for the issue: [Expected indentation of 8 spaces but found 3 tabs.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
        errorHandler: false,
```

File: 1host.config.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+	port: 3000,
+	modules: [
+		{
+			module: require("./index.js"),
```
Comment (RedGuy12) [inline 1host.config.js:5]: Codacy has a fix for the issue: [Do not reference the index file directly..](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
			module: require('.'),
```

File: 1host.config.js (hunk 2 above)
Comment (RedGuy12) [inline 1host.config.js:7]: Codacy has a fix for the issue: [Unexpected trailing comma.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
			config: { usernamee: "wgyt", projectid: "513842800", commentid: "200363559" }
```

File: 1host.config.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+	port: 3000,
+	modules: [
```
Comment (RedGuy12) [inline 1host.config.js:3]: Codacy has a fix for the issue: [Expected indentation of 2 spaces but found 1 tab.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
  modules: [
```

File: 1host.config.js (hunk 4 above)
Comment (RedGuy12) [inline 1host.config.js:5]: Codacy has a fix for the issue: [Expected indentation of 6 spaces but found 3 tabs.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
      module: require("./index.js"),
```

File: 1host.config.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+		{
```
Comment (RedGuy12) [inline 1host.config.js:4]: Codacy has a fix for the issue: [Expected indentation of 4 spaces but found 2 tabs.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
    {
```

File: 1host.config.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+			module: require("./index.js"),
+			errorHandler: false,
+			config: { usernamee: "wgyt", projectid: "513842800", commentid: "200363559" },
+		},
```
Comment (RedGuy12) [inline 1host.config.js:8]: Codacy has a fix for the issue: [Expected indentation of 4 spaces but found 2 tabs.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
    },
```

File: 1host.config.js (hunk 2 above)
Comment (RedGuy12) [inline 1host.config.js:7]: Codacy has a fix for the issue: [Unquoted property 'config' found.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
			"config": { usernamee: "wgyt", projectid: "513842800", commentid: "200363559" },
```

File: index.js [hunk 8]
```diff
@@ -78,20 +71,40 @@ module.exports = async (req, res, config) => {
...
+			if (comment.value == key) {
+				var uname = comment.user;
+				var verif = true;
+				let data = `username=${uname}&verified=${verif}&ocular=https%3A%2F%2Fmy-ocular.jeffalo.net%2Fapi%2Fuser%2F${uname}&scratchdb=https%3A%2F%2Fscratchdb.lefty.one%2Fv2%2Fuser%2Finfo%2F${uname}`;
+				if (queryString.url.includes("?")) {
+					data = `&` + data;
+				} else {
+					data = `?` + data;
 				}
+				return res.start(`<meta http-equiv="refresh" content="0; url=${queryString.url}${data}"/> HI`,'text/html');
```
Comment (wgyt) [inline index.js:104]: still not working, this line hasn't been working for some reason

File: index.js [hunk 9]
```diff
@@ -78,20 +71,40 @@ module.exports = async (req, res, config) => {
...
+				if (queryString.url.includes("?")) {
 					data = `&` + data;
 				} else {
 					data = `?` + data;
 				}
 				if (fresh) {
 					await db.set(key, false);
-					res.writeHead(302, { Location: `${req.query.url}${data}` });
-					return res.start();
+					return res.start(`<meta http-equiv="refresh" content="0; url=${queryString.url}${data}"/>`);
```
Comment (wgyt) [inline index.js:81]: also this line doesn't

File: index.js [hunk 10]
```diff
@@ -78,20 +71,40 @@ module.exports = async (req, res, config) => {
...
 				} else {
-					res.writeHead(302, { Location: `/?url=${req.query.url}` });
-					return res.start();
+					return res.start(`<meta http-equiv="refresh" content="0; url=/?url=${queryString.url}"/>`);
```
Comment (wgyt) [inline index.js:83]: and this line

**Comment by wgyt** [comment]
seems like 1host errors
could @GrahamSH-LLK help fix this?

File: index.js [hunk 11]
```diff
@@ -5,46 +5,39 @@ module.exports = async (req, res, config) => {
...
-					{ code: code, url: url },
-					{},
-					function (err, str) {
-						if (err) {
-							throw err;
-						}
-						text = str;
-					},
-				);
+				ejs.renderFile(`./pages/authwith${cloud}.ejs`, { code: code, url: url }, {}, function (err, str) {
```
Comment (RedGuy12) [inline index.js:21]: Codacy has a fix for the issue: [Expected property shorthand.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
				ejs.renderFile(`./pages/authwith${cloud}.ejs`, { code, url: url }, {}, function (err, str) {
```

File: 1host.config.js [hunk 12]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+	port: 3000,
+  modules: [
+    {
+      module: require("./index.js"),
+        errorHandler: false,
+			config: { usernamee: "wgyt", projectid: "513842800", commentid: "200363559" }
```
Comment (RedGuy12) [inline 1host.config.js:7]: Codacy has a fix for the issue: [Expected indentation of 6 spaces but found 3 tabs.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
      config: { usernamee: "wgyt", projectid: "513842800", commentid: "200363559" }
```

File: 1host.config.js [hunk 13]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+	port: 3000,
+  modules: [
+    {
+      module: require("./index.js"),
+        errorHandler: false,
```
Comment (RedGuy12) [inline 1host.config.js:6]: Codacy has a fix for the issue: [Unquoted property 'errorHandler' found.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
			"errorHandler": false,
```

File: 1host.config.js (hunk 12 above)
Comment (RedGuy12) [inline 1host.config.js:7]: Codacy has a fix for the issue: [Expected indentation of 8 spaces but found 3 tabs.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
        config: { usernamee: "wgyt", projectid: "513842800", commentid: "200363559" }
```

File: 1host.config.js [hunk 14]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+	port: 3000,
+  modules: [
+    {
+      module: require("./index.js"),
```
Comment (RedGuy12) [inline 1host.config.js:5]: Codacy has a fix for the issue: [Do not reference the index file directly..](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
			module: require('.'),
```

File: 1host.config.js [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+        errorHandler: false,
+			config: { usernamee: "wgyt", projectid: "513842800", commentid: "200363559" }
+    },
+		{ module: require("@1hostjs/errors"), errorHandler: true, config:{
```
Comment (RedGuy12) [inline 1host.config.js:9]: Codacy has a fix for the issue: [Missing space before value for key 'config'.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
		{ module: require("@1hostjs/errors"), errorHandler: true, config: {
```

File: 1host.config.js [hunk 16]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+	port: 3000,
+  modules: [
+    {
+      module: require("./index.js"),
+        errorHandler: false,
+			config: { usernamee: "wgyt", projectid: "513842800", commentid: "200363559" }
+    },
```
Comment (RedGuy12) [inline 1host.config.js:8]: Codacy has a fix for the issue: [Expected indentation of 4 spaces but found 2 tabs.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
    },
```

File: 1host.config.js (hunk 14 above)
Comment (RedGuy12) [inline 1host.config.js:5]: Codacy has a fix for the issue: [Expected indentation of 8 spaces but found 3 tabs.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
        module: require("./index.js"),
```

File: 1host.config.js (hunk 15 above)
Comment (RedGuy12) [inline 1host.config.js:9]: Codacy has a fix for the issue: [Expected indentation of 6 spaces but found 2 tabs.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
      { module: require("@1hostjs/errors"), errorHandler: true, config:{
```

File: 1host.config.js (hunk 14 above)
Comment (wgyt) [reply 1host.config.js:5]: not committing this suggestion because 1host needs a file path

File: 1host.config.js [hunk 17]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+	port: 3000,
+  modules: [
+    {
+        module: require("./index.js"),
+			"errorHandler": false,
+      config: { usernamee: "wgyt", projectid: "513842800", commentid: "200363559" }
```
Comment (RedGuy12) [inline 1host.config.js:7]: Codacy has a fix for the issue: [Insert `,`](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
			config: { usernamee: "wgyt", projectid: "513842800", commentid: "200363559" },
```

File: 1host.config.js [hunk 18]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+	port: 3000,
+  modules: [
+    {
+        module: require("./index.js"),
```
Comment (RedGuy12) [inline 1host.config.js:5]: Codacy has a fix for the issue: [Expected indentation of 8 spaces but found 3 tabs.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
        module: require("./index.js"),
```

File: 1host.config.js (hunk 17 above)
Comment (RedGuy12) [inline 1host.config.js:7]: Codacy has a fix for the issue: [Expected indentation of 8 spaces but found 3 tabs.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
        config: { usernamee: "wgyt", projectid: "513842800", commentid: "200363559" }
```

File: 1host.config.js [hunk 19]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+	port: 3000,
+  modules: [
```
Comment (RedGuy12) [inline 1host.config.js:3]: Codacy has a fix for the issue: [Replace `··` with `↹`](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
	modules: [
```

File: 1host.config.js (hunk 18 above)
Comment (RedGuy12) [inline 1host.config.js:5]: Codacy has a fix for the issue: [Expected indentation of 6 spaces but found 3 tabs.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
      module: require("./index.js"),
```

File: 1host.config.js [hunk 20]
```diff
@@ -0,0 +1,14 @@
...
+    {
+        module: require("./index.js"),
+			"errorHandler": false,
```
Comment (RedGuy12) [inline 1host.config.js:6]: Codacy has a fix for the issue: [Replace `"errorHandler"` with `errorHandler`](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
			errorHandler: false,
```

File: 1host.config.js [hunk 21]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+	port: 3000,
+  modules: [
+    {
```
Comment (RedGuy12) [inline 1host.config.js:4]: Codacy has a fix for the issue: [Replace `····` with `↹↹`](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
		{
```

File: 1host.config.js [hunk 22]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+	port: 3000,
```
Comment (RedGuy12) [inline 1host.config.js:2]: Codacy has a fix for the issue: [Expected indentation of 2 spaces but found 1 tab.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
  port: 3000,
```

File: 1host.config.js (hunk 17 above)
Comment (RedGuy12) [inline 1host.config.js:7]: Codacy has a fix for the issue: [Object properties must go on a new line.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
			config: { usernamee: "wgyt",
```

File: 1host.config.js [hunk 23]
```diff
@@ -0,0 +1,14 @@
...
+  modules: [
+    {
+        module: require("./index.js"),
+			"errorHandler": false,
+      config: { usernamee: "wgyt", projectid: "513842800", commentid: "200363559" }
+    },
+		{ module: require("@1hostjs/errors"), errorHandler: true, config: {
```
Comment (RedGuy12) [inline 1host.config.js:9]: Codacy has a fix for the issue: [Unquoted property 'module' found.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
		{ "module": require("@1hostjs/errors"), errorHandler: true, config: {
```

File: 1host.config.js (hunk 17 above)
Comment (RedGuy12) [inline 1host.config.js:7]: Codacy has a fix for the issue: [Unquoted property 'config' found.](https://app.codacy.com/gh/onedotprojects/scratchCommentAuth/pullRequest?prid=7364619)
```suggestion
			"config": { usernamee: "wgyt", projectid: "513842800", commentid: "200363559" }
```

**Comment by FunctionalMetatable** [comment]
_if 1host doesnt work, remove 1host_

**Comment by wgyt** [comment]
> _if 1host doesnt work, remove 1host_

_1host works on the latest git commit, but it isn't published to npm yet_

**Comment by wgyt** [comment]
As long as I fix the bug later, I’ll merge

**Pull request merged**